core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/http/pipeline/stages/AsyncRetryableStage.java

```diff
@@ -231,4 +261,16 @@ private int readLimit() {
             return RequestOption.DEFAULT_STREAM_BUFFER_SIZE;
         }
     }
+
+    private long getClientExecutionTimeoutInMillis(RequestOverrideConfiguration requestConfig) {
+        if (requestConfig.apiCallTimeout().isPresent()) {
+            return requestConfig.apiCallTimeout().get().toMillis();
+        }
+
+        if (clientConfig.option(SdkClientOption.API_CALL_TIMEOUT) != null) {
```

Comment:
Can we use `OptionalUtils.firstPresent`?

Reply:
+1, was not aware we have those fancy utils